replica/table.cc

```diff
@@ -2302,21 +2302,30 @@ future<row_locker::lock_holder> table::do_push_view_replica_updates(schema_ptr s
         co_return row_locker::lock_holder();
     }
     auto cr_ranges = co_await db::view::calculate_affected_clustering_ranges(*base, m.decorated_key(), m.partition(), views);
-    if (cr_ranges.empty()) {
+    const bool need_regular = !cr_ranges.empty();
+    const bool need_static = db::view::needs_static_row(m.partition(), views);
+    if (!need_regular && !need_static) {
         tracing::trace(tr_state, "View updates do not require read-before-write");
         co_await generate_and_propagate_view_updates(base, sem.make_tracking_only_permit(s.get(), "push-view-updates-1", timeout), std::move(views), std::move(m), { }, std::move(tr_state), now);
         // In this case we are not doing a read-before-write, just a
         // write, so no lock is needed.
         co_return row_locker::lock_holder();
     }
-    // We read the whole set of regular columns in case the update now causes a base row to pass
+    // We read the whole set of regular and static columns in case the update now causes a base row to pass
     // a view's filters, and a view happens to include columns that have no value in this update.
     // Also, one of those columns can determine the lifetime of the base row, if it has a TTL.
-    auto columns = boost::copy_range<query::column_id_vector>(
-            base->regular_columns() | boost::adaptors::transformed(std::mem_fn(&column_definition::id)));
+    query::column_id_vector columns;
+    if (need_regular) {
+        boost::copy(base->regular_columns() | boost::adaptors::transformed(std::mem_fn(&column_definition::id)), std::back_inserter(columns));
+    }
+    if (need_static) {
```

Comment:
it might be a totally unimportant detail, but isn't the usual order (partition key, clustering key, static columns, regular columns)? In which case, you might want to put this `if` condition above the one for need_regular. I have no idea if we ever rely on this order, but maybe some code paths do

Reply:
Actually, it turns out that `query::partition_slice()` takes two separate arguments of type `query::column_id_vector`, one for static and the other for regular columns :) I didn't notice it before because the variable was just named `columns` which suggested that the split isn't needed, and everything seemed to work correctly despite that.

I fixed it now so that static and regular columns are passed separately.